tests/functional/modules/test_zos_encode_func.py

```diff
@@ -77,25 +77,60 @@
 /*
 """
 
+VSAM_RECORDS = """00000001A record
+00000002A record
+00000003A record
+"""
+
+def create_vsam_data_set(hosts, name, ds_type, add_data=False, key_length=None, key_offset=None):
+    """Creates a new VSAM on the system.
+
+    Arguments:
+        hosts (object) -- Ansible instance(s) that can call modules.
+        name (str) -- Name of the VSAM data set.
+        type (str) -- Type of the VSAM (KSDS, ESDS, RRDS, LDS)
+        add_data (bool, optional) -- Whether to add records to the VSAM.
+        key_length (int, optional) -- Key length (only for KSDS data sets).
+        key_offset (int, optional) -- Key offset (only for KSDS data sets).
+    """
+    params = dict(
+        name=name,
+        type=ds_type,
+        state="present"
+    )
+    if ds_type == "KSDS":
+        params["key_length"] = key_length
+        params["key_offset"] = key_offset
+
+    hosts.all.zos_data_set(**params)
+
+    if add_data:
+        record_src = "/tmp/zos_copy_vsam_record"
+
+        hosts.all.zos_copy(content=VSAM_RECORDS, dest=record_src)
```

Comment:
I think I mentioned this in another PR, but we want to avoid introducing new dependencies between modules in the test cases, we can use `hosts.all.shell` to do this with `echo` instead of using `zos_copy`

Reply:
Is better to fix this now instead of having to do another round of development before merging into `dev`

Reply:
done